checkov/terraform/checks/resource/azure/AKSApiServerAuthorizedIpRanges.py

```diff
@@ -1,23 +1,23 @@
-from typing import List, Any
-
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
 from checkov.common.models.consts import ANY_VALUE
-from checkov.common.models.enums import CheckCategories
-from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
 
 
-class AKSApiServerAuthorizedIpRanges(BaseResourceValueCheck):
+class AKSApiServerAuthorizedIpRanges(BaseResourceCheck):
     def __init__(self) -> None:
         name = "Ensure AKS has an API Server Authorized IP Ranges enabled"
         id = "CKV_AZURE_6"
         supported_resources = ("azurerm_kubernetes_cluster",)
         categories = (CheckCategories.KUBERNETES,)
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def get_inspected_key(self) -> str:
-        return "api_server_authorized_ip_ranges/[0]"
-
-    def get_expected_values(self) -> List[Any]:
-        return [ANY_VALUE]
-
+    def scan_resource_conf(self, conf):
+        private_cluster_enabled = conf.get('private_cluster_enabled',[False])[0]
+        api_server_authorized_ip_ranges =  conf.get('api_server_authorized_ip_ranges',[[]])[0]
+     
+        if not private_cluster_enabled:
+            if api_server_authorized_ip_ranges == []:
+                return CheckResult.FAILED
+        return CheckResult.PASSED
```

Comment:
```suggestion
    def scan_resource_conf(self, conf):
        # can't be set for private cluster
        private_cluster_enabled = conf.get('private_cluster_enabled',[False])[0]
        if private_cluster_enabled:
            return CheckResult.PASSED
        return super().scan_resource_conf(conf)
```
you can something like this and leave the old code like it was 🙂

Reply:
Thanks, added the scan_resource_conf() as you suggested!

Example and test are also updated